lms/models/lti_user.py

```diff
@@ -59,6 +66,15 @@ def from_auth_params(application_instance, lti_core_schema):
             email=lti_core_schema["lis_person_contact_email_primary"],
         )
 
+    @staticmethod
+    def unserialize(request, **kwargs: dict):
+        # pylint: disable= import-outside-toplevel
+        from lms.services.lti_role_service import LTIRoleService
+
+        lti_roles = request.find_service(LTIRoleService).get_roles(kwargs["roles"])
```

Comment:
This really screams to me that something here isn't quite right. I don't think our models should be calling services. The import here is something that can probably be worked around, but I suspect is indicating the same.

I like keeping the serialization and deserializaton next to each other, but this seems to lead to either both going to a service, or both coming here (but pulling in a service). Both bad in their own way.

The minimal tweak would be to inject the service as an argument. We already inject the request, and we only use it to get the service, but I do think it's telling us something bigger isn't right.